### PR TITLE
Refactor get

### PR DESCRIPTION
**Pull request by sitaram-kalluri**
**- What I did** Refactor get method for readability.

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: at_client/lib/src/client/local_secondary.dart [hunk 1]
```diff
@@ -120,6 +120,9 @@ class LocalSecondary implements Secondary {
       if (builder.isCached) {
         llookupKey += 'cached:';
       }
+      if(builder.isPublic){
+        llookupKey += 'public:';
+      }
       if (builder.sharedWith != null) {
         llookupKey += '${AtUtils.formatAtSign(builder.sharedWith!)}:';
       }
```
Comment (gkc) [inline at_client/lib/src/client/local_secondary.dart:128]: Are cached, public and sharedWith all mutually exclusive? If so, perhaps have some defensive code in here to check that at most *one* of these three booleans is true?

File: at_client/lib/src/util/at_client_validation.dart [hunk 2]
```diff
@@ -52,4 +52,18 @@ class AtClientValidation {
...
   }
+
+  static void validateAtKey(AtKey atKey) {
+    // validates the atKey
+    validateKey(atKey.key);
+    // validates the metadata
+    validateMetadata(atKey.metadata);
+    // verifies if the sharedWith atSign exists.
+    if (atKey.sharedWith != null) {
+      isAtSignExists(
```
Comment (gkc) [inline at_client/lib/src/util/at_client_validation.dart:65]: If disconnected from the network, isAtSignExists will throw an exception. In this context (validation), will the exception message make sense to a user if for example the app displays the message on screen?

File: at_client/lib/src/client/verb_builder_manager.dart [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+class SecondaryManager {
+  static Secondary getSecondary(VerbBuilder verbBuilder) {
+    if (verbBuilder is LookupVerbBuilder ||
+        verbBuilder is PLookupVerbBuilder ||
+        verbBuilder is NotifyVerbBuilder ||
+        verbBuilder is StatsVerbBuilder) {
+      return AtClientManager.getInstance().atClient.getRemoteSecondary()!;
+    }
+    return AtClientManager.getInstance().atClient.getLocalSecondary()!;
```
Comment (gkc) [inline at_client/lib/src/client/verb_builder_manager.dart:81]: Is there an explanation somewhere of which verbs need to be executed on local secondary vs remote secondary?

**Review by gkc — COMMENTED** [review]
LGTM, apart from a few comments / requests

At some point in the future we need to come back and think about this whole `AtClientManager.getInstance()` business again :-)

File: at_client/lib/src/client/local_secondary.dart (hunk 1 above)
Comment (sitaram-kalluri) [reply at_client/lib/src/client/local_secondary.dart:128]: public and sharedWith are mutually exclusive(not cached). However there can be a key without any of these(for exmaple self keys and encryption keys).

**Review by gkc — APPROVED** [review]

File: at_client/lib/src/util/at_client_validation.dart (hunk 2 above)
Comment (sitaram-kalluri) [reply at_client/lib/src/util/at_client_validation.dart:65]: Updated the code to perform this check only network connection is available.

File: at_client/lib/src/client/verb_builder_manager.dart (hunk 3 above)
Comment (sitaram-kalluri) [reply at_client/lib/src/client/verb_builder_manager.dart:81]: Updated the dart docs

**Pull request merged**